### PR TITLE
display livepatch support

### PR DESCRIPTION
**Pull request by orndorffgrant**
## New Notes

Fixes: #913

TODO
- [x] move cache file definition to state_files.py

## Notes

While starting this I've noticed some features to break out into separate PRs and merge beforehand. They are:

- [x] #2168 move system query functions from `util.py` to a new `system.py` because I'll need to add more here and util is already getting large. This split will help organize things
- [x] #2184 refactor how we get and parse the kernel version. livepatch uses `/proc/version_signature` and needs more granular version information (specifically, the hwe revision). We need to parse the version in the same way to properly query for kernel support.
- [x] #2189 add a new function to get the arch from `lscpu`. This is the value that livepatch uses and could be slightly different from what dpkg reports.
- [x] #2169 `DateDataValue` for parsing date strings into DataObjects
- [x] #2172 `DataObjectFile`. For the cache, it will be useful to have an file object that automatically serializes/parses content to/from a DataObject. I'll make this reuse-able.

From there I think I'll have the important building blocks for this feature and the rest will probably be in this PR.

## Test Steps
- run the new integration test
- start a vm and try out the UX for an unsupported vs supported kernel when attached/unattached/enabled/disabled
- check the logs to verify the api/cache/cli are used for support information appropriately (search for "livepatch.py")

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/entitlements/livepatch.py [hunk 1]
```diff
@@ -26,9 +38,235 @@
...
+            if cli_supported == "unsupported":
+                return False
+
+            cli_state = cli_status.get("State", None)
+            if cli_state == "applied":
+                logging.debug(
+                    "kernel must be supported because livepatches "
+                    " have been applied"
+                )
+                return True
```
Comment (orndorffgrant) [inline uaclient/entitlements/livepatch.py:155]: We actually may not be able to assume this means the kernel is supported. For example, if a kernel is supported for a while but then we end support, then we need to warn the user that their support has ended even if the kernel has patches currently applied.

**Review by renanrodrigo — APPROVED** [review]
this lgtm

**Comment by orndorffgrant** [comment]
The cloud test failures indicate there may be a gap in the support database

**Comment by renanrodrigo** [comment]
@orndorffgrant if we confirm that the cloud kernels are now right in the database, our tests are looking good and I guess we can click the magic button here

**Comment by orndorffgrant** [comment]
Cloud kernel support is updated in the database and just rebased and resolved conflicts. Let's let CI run

**Comment by renanrodrigo** [comment]
MERGE, MERGE EVERYTHING

**Pull request merged**